### DISCUSSION TITLE
OpenShift Virtualization Provider supported?

### DISCUSSION THREAD

**Discussion by marcoshaw** [crc-org/crc#3387] [Q&A] [answered]
EDIT: I'm running 4.11.3 on Windows.

Should this provider work within CRC?

I have:
-Windows 11 Pro (I recently upgraded to Pro just for this)

I have tried:
-4.11.5: Windows 11 Pro, Hyper-V, CRC
-4.11.7 (nested): Windows 11 Pro, VMware Workstation Pro 16, RHEL 9, CRC

In both cases, once I try to add the virtualization provider, the system seems to suddenly become unstable each time.  I'm never even able to see the provider finish/complete to even try to load a VM.

I've tried upping to 8 CPUs and 32GB of memory and it still seems to fail.

I guess crc logging only seems to cover the start up of the VM.  Perhaps I need to get into the crc VM to actually try to find any relevant logs, assuming it "should work"?

**Comment by praveenkumar** [comment]
@marcoshaw can you let us know which document are you following to enable virtualization provider?

**Comment by marcoshaw** [comment]
This would be how I try it:
https://docs.openshift.com/container-platform/4.11/virt/install/installing-virt-web.html

As for step #7, I've tried it at different times as it's not clear (to me) when one should actually do it, if it matters.

As for running it in a nested RHEL 9 VM, I understand that KVM nested virtualization is in "tech preview" only (per Red Hat https://access.redhat.com/solutions/6692341), but I'm not even at the point of trying to install a VM.

EDIT: I'm a bit unsure how/where to make the Red Hat recommendations for KVM nesting.  I'll dig a bit.

**Comment by marcoshaw** [comment]
https://github.com/code-ready/crc/issues/3334#issuecomment-1238863738

I never noticed that nesting wasn't supported.  It would be my preference, but I understand it's potentially extra work/debugging.

I may go back to trying this in Hyper-V and see if I can get inside the VM and look around.

Beyond that, I may just try to drop a few bucks and try to run this on Azure just so I can be satisfied that I tried it.  I don't know.

**Comment by marcoshaw** [comment]
Some better news to report...

I forced 4.11.7 to install on my Windows/HyperV, and assigned 8 CPU and 32GB of RAM, and the virtualization operator installed within a few minutes.

I'll loading a VM image in the next day or so.

**Comment by marcoshaw** [comment]
I see this series of events every 5 minutes (see image).  I tried to create a VM, but failed.

![error](https://user-images.githubusercontent.com/2267513/196818807-4c2e0eb0-dde9-4c95-8d92-504d828dde66.png)

**Comment by praveenkumar** [comment, marked as answer]
https://docs.openshift.com/container-platform/4.11/virt/install/preparing-cluster-for-virt.html#preparing-cluster-for-virt check the requirements I think you have crc running on the hyperV and now hyperV suppose to support nested virtualization. I think if you really want to try crc with virt support better to do it on Linux host (mostly baremetal).

**Comment by marcoshaw** [comment]
I've now setup a bare metal RHEL 8 for this.  The operator finally seems to install (and be stable).  Now it appears there's some manual configuration for me to do (example https://myopenshiftblog.com/example-of-deploying-a-windows-vm-from-a-template-using-openshift-virtualization/).

I loaded an ISO in PVC (if that's the write term), and tried to create a VM from a template, but it won't start, so I'm guessing I need to fiddle with disk storage (and possibly networking).
